src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,25 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate){
+    	  this.isPrivate = isPrivate;
```

Comment:
You may also extract another method from the child classes, that takes in a regex string, and perform the match.